packages/core/src/data-editor/data-editor.tsx

```diff
@@ -1527,6 +1530,7 @@ const DataEditorImpl: React.ForwardRefRenderFunction<DataEditorRef, DataEditorPr
                     setGridSelection(undefined, false);
                     setSelectedRows(CompactSelection.empty());
                     setSelectedColumns(CompactSelection.empty(), "selection");
+                    onSelectionCleared?.();
```

Comment:
Shouldn't we call this when we click out-of-bounds as well?
Around here:
https://github.com/glideapps/glide-data-grid/blob/6b857686c061ca8d9e3fffca4135356625b4ac63/packages/core/src/data-editor/data-editor.tsx#L939

Reply:
Good catch